src/Marten/Events/Daemon/ActionParameters.cs

```diff
@@ -54,15 +54,16 @@ public void IncrementAttempts(TimeSpan delay = default)
             Cancellation = Group.Cancellation;
         }
 
-        public void ApplySkip(SkipEvent skip)
+        public async ValueTask ApplySkip(SkipEvent skip)
```

Comment:
@oskardudycz,  I inspected all downstream calls: 4 of 5  run synchronously  except `ViewProjection.SliceAsyncEvents()`. The last has sync branches too.  According to these facts I believe it's better to keep `ValueTask` as a return value but rename to `ApplySkipAsync()`. 
It's possible to micro-optimize it as shown below but I doubt it worse 3 lines of code 

```csharp
        public async ValueTask ApplySkipAsync(SkipEvent skip)
        {
            if (Group != null)
            {
                ValueTask skipTask = Group.SkipEventSequence(skip.Event.Sequence);
                if (!skipTask.IsCompleted)
                {
                    await skipTask.ConfigureAwait(false);
                }

                // You have to reset the CancellationToken for the group
                Group.Reset();
                Cancellation = Group.Cancellation;
            }

            // Basically saying that the attempts start over when we skip
            Attempts = 0;
            Delay = default;
        }
```

Reply:
Hmm, I might be missing something, but isn't it only used in Projection Daemon (precisely: https://github.com/JasperFx/marten/blob/master/src/Marten/Events/Daemon/ProjectionDaemon.cs#L402)? It's already an asynchronous method, so we should be good. Without that, all the sync methods would need to call `.Wait()` to be sure that there are no threading issues.

Thoughts?

Reply:
Yes, `ActionParameters.SkipAsync()` is used in `ProjectionDaemon.TryAction()` only. Further calls may be async or sync (sync by nature, not need to `Task.Wait()` calls). Pseudo-graphics below shows possible call trees.
async Task ProjectionDaemon.TryAction() 
---> async ValueTask ActionParameters.ApplySkipAsync()
-----> abstract ValueTask [EventRangeGroup.SkipEventSequence()
------|=> async ValueTask [TenantSliceRange<TDoc, TId>.SkipEventSequence()](https://github.com/JasperFx/marten/blob/b5325056d6226159565036b0eda89d0b89ff5429/src/Marten/Events/Aggregation/TenantSliceRange.cs#L57)
---------> abstract ValueTask IEventSlicer.SliceAsyncEvents()
----------|=> sync ValueTask [ByStreamId.SliceAsyncEvents()](https://github.com/JasperFx/marten/blob/08efa3cfb31c3549f2449bb7da5cfd70a0bb2eb1/src/Marten/Events/Aggregation/ByStreamId.cs#L22)
----------|=> sync ValueTask [ByStreamKey<TDoc>.SliceAsyncEvents()](https://github.com/JasperFx/marten/blob/08efa3cfb31c3549f2449bb7da5cfd70a0bb2eb1/src/Marten/Events/Aggregation/ByStreamKey.cs#L20)
----------|=> async ValueTask [ViewProjection<TDoc, TId>.SliceAsyncEvents()](https://github.com/JasperFx/marten/blob/b5325056d6226159565036b0eda89d0b89ff5429/src/Marten/Events/Projections/ViewProjection.cs#L65)
------|=>--sync ValueTask [TenantedEventRange<TDoc, TId>.SkipEventSequence()](https://github.com/JasperFx/marten/blob/df12e5a065b82b9edd0497e9a08d2faedb7f5a33/src/Marten/Events/Daemon/TenantedEventRange.cs#L46)

I still think making signature `public async ValueTask ApplySkipAsync(SkipEvent skip)` is reasonable as it can be completed in a thread safe and synchronous way.

Reply:
I'm not an async expert, but in my opinion, it's more on how do we intend to use it, not what methods we're using inside. 
At this stage, we already know that `ApplySkipAsync` is not used anywhere, and we do not intend to use it synchronously. Thus it'd be more straightforward to state that by the method's signature explicitly. When we have the need to use it synchronously (hopefully never 😅), we can change it to ValueTask.

It's not a blocker for me tho.

Reply:
`ApplySkipAsync` is always async in terms of usage and not going to be used synchronously. Hence agree with @oskardudycz that it is better for the return type to be `Task` which is more clear rather than `ValueTask`. That said, the class is `internal` so it is not a blocker by any means.

Reply:
Ok, I have changed it to `Task`.